### PR TITLE
Testing comment trigger for actions

### PR DESCRIPTION
**Pull request by pogil**
## What changes are proposed in this pull request?

Add a check to trigger lint by adding a comment containing `trigger lint`.

## How is this patch tested?

This will be the test. 

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

Just a test, won't be committed.

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -5,10 +5,15 @@ on:
...
     branches: [ master ]
+  issue_comment:
+    types: [ created ]
+    
+    
 
 jobs:
   lint:
     runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request' || github.event_name == 'push' || (github.event_name == 'issue_comment' && contains(github.comment.html_url, '/pull/') && contains(github.event.comment.body, 'trigger lint')) }}
```
Comment (pogil) [inline .github/workflows/master.yml:16]: We would need to add similar lines for other jobs. Unfortunately docs suggest you can't do an `if` in the `on` block, nor can you an `if` on the list of all jobs.

**Comment by pogil** [comment]
trigger lint

**Comment by pogil** [comment]
trigger docs

**Comment by pogil** [comment]
trigger lint

**Comment by harupy** [comment]
I found a related post: https://github.community/t/trigger-action-via-pull-request-message/17630. It seems pretty complicated to achieve this feature.

**Comment by harupy** [comment]
We could use the [re-run-workflow](https://developer.github.com/v3/actions/workflow-runs/#re-run-a-workflow) API (note that this API re-runs the whole workflow and has no option which allows us to re-run a specific job). This approach would be like:

1. Use `issue_comment` event as a trigger.
2. Find the id of a workflow that we want to re-run (not sure if this is really possible)
3. Hit the re-run-workflow API with the workflow id found in the previous step.

**Comment by pogil** [comment]
@harupy thanks for doing the research here! I had came to a similar conclusion but hadn't found the re-run-workflow API yet. I'll take a look

**Comment by pogil** [comment]
1. Get pull request of issue comment from `github.event.issue.pull_request.url` in workflow for issue_comment type actions
2. Look up that pull request
> curl 'https://api.github.com/repos/mlflow/mlflow/pulls/2930'
3. Get version from the pull request from 'head.sha' in JSON
4. Use the sha to look up check suites for that commit
> curl -H'Accept: application/vnd.github.antiope-preview+json' 'https://api.github.com/repos/mlflow/mlflow/commits/cbb29bc1aa61c8e7ed4c53aab16b1fc519c2f581/check-suites'

^^ this could in theory return multiple check suites, but in practice it may be broken. There's a valid scenario where you may open the same pull requests against different branches. It looks like in the UI with a test I did the check suites get shared across the two branches rather than trying the merge request for each branch separately. That simplifies bug validation, I guess? It's kind hard to know from information in the check suite which branch it was run on, because it just has the before/after commit hashes. 

5. Trigger a rerun using the check_suites `id`:
> curl -uuser:token -X POST -H'Accept: application/vnd.github.antiope-preview+json' 'https://api.github.com/repos/mlflow/mlflow/check-suites/791794316/rerequest'

**Comment by pogil** [comment]
trigger checks

**Comment by harupy** [comment]
It seems we need to merge the PR to the master branch to enable this action. [The doc](https://help.github.com/en/actions/reference/events-that-trigger-workflows#issue-comment-event-issue_comment) says:

> Note: This event (`issue_comment`) will only trigger a workflow run if the workflow file is on the master or default branch.

**Comment by harupy** [comment]
When I ran the check-suite re-request API **on GitHub Action**, it didn't work (tested on [my dummy repo](https://github.com/harupy/trigger-action-by-comment)). The check-suite was successfully re-requested, but its status never changes from "queued" and the GitHub UI doesn't seem to update.

Screen Shot 2020-06-17 at 22 15 14

https://github.com/harupy/trigger-action-by-comment/pull/2/checks?check_run_id=779661702

**Comment by pogil** [comment]
Yeah I expect I will hit a dead end using GitHub for this due to permissions. We could put a token in a secret, but then the commenter would have to have edit access to the workflow file to read the secret during workflow execution if I read the docs right.

Looks like the only way might be some externally hosted service that + webhooks

**Comment by harupy** [comment]
Or make our own bot using GitHub Apps and give it permission required to do the job.

**Comment by harupy** [comment]
I used my personal token instead of `secrets.GITHUB_TOKEN` and it worked.

**Comment by harupy** [comment]
GitHub Apps don't seem to work either. As reported [here](https://github.com/probot/probot/issues/1065#issuecomment-576444183), check-suites created by GitHub Actions can't be updated from other GitHub Apps.

**Comment by harupy** [comment]
@pogil Would it be useful to introduce a bot that automatically adds labels based on a PR description?

**Pull request closed without merging**